File: JavaServletSrc/arfn_confirm_doc.js
function arfn_confirm_doc() {
var pID;
var pTipo;
var pDocUnico;
var pDoc1;
var pDoc2;
var pDoc3;
var pDoc4;
var pDoc5;
var pDoc6;
var pOld0;
var pOld1;
var pOld2;
var pOld3;
var pOld4;
var pOld5;
var pOld6;
var pAttivi;
var pTicket;
var pCodDip;
var m_Caller
arfn_confirm_doc._p=_rargs(arguments,{eval:function(e){eval(e)}})
function i_BlankParameters() {
pID='';
pTipo='';
pDocUnico='';
pDoc1='';
pDoc2='';
pDoc3='';
pDoc4='';
pDoc5='';
pDoc6='';
pOld0='';
pOld1='';
pOld2='';
pOld3='';
pOld4='';
pOld5='';
pOld6='';
pAttivi='';
pTicket='';
pCodDip='';
}
if(arfn_confirm_doc._p.length>0)_rargs([[['pID',arfn_confirm_doc._p[0]],['pTipo',arfn_confirm_doc._p[1]],['pDocUnico',arfn_confirm_doc._p[2]],['pDoc1',arfn_confirm_doc._p[3]],['pDoc2',arfn_confirm_doc._p[4]],['pDoc3',arfn_confirm_doc._p[5]],['pDoc4',arfn_confirm_doc._p[6]],['pDoc5',arfn_confirm_doc._p[7]],['pDoc6',arfn_confirm_doc._p[8]],['pOld0',arfn_confirm_doc._p[9]],['pOld1',arfn_confirm_doc._p[10]],['pOld2',arfn_confirm_doc._p[11]],['pOld3',arfn_confirm_doc._p[12]],['pOld4',arfn_confirm_doc._p[13]],['pOld5',arfn_confirm_doc._p[14]],['pOld6',arfn_confirm_doc._p[15]],['pAttivi',arfn_confirm_doc._p[16]],['pTicket',arfn_confirm_doc._p[17]],['pCodDip',arfn_confirm_doc._p[18]]]],{eval:function(e){eval(e)}})
arfn_confirm_doc._s=new BatchJavascript();
if (Ne(typeof(document.FSender),'undefined') && Ne(typeof(document.FSender.m_cRegionalSettings),'undefined')) {
arfn_confirm_doc._s.SetParameterString('m_cRegionalSettings',document.FSender.m_cRegionalSettings.value);
}
arfn_confirm_doc._s.SetParameterString('pID',WtA(pID,'C'));
arfn_confirm_doc._s.SetParameterString('pTipo',WtA(pTipo,'C'));
arfn_confirm_doc._s.SetParameterString('pDocUnico',WtA(pDocUnico,'C'));
arfn_confirm_doc._s.SetParameterString('pDoc1',WtA(pDoc1,'C'));
arfn_confirm_doc._s.SetParameterString('pDoc2',WtA(pDoc2,'C'));
arfn_confirm_doc._s.SetParameterString('pDoc3',WtA(pDoc3,'C'));
arfn_confirm_doc._s.SetParameterString('pDoc4',WtA(pDoc4,'C'));
arfn_confirm_doc._s.SetParameterString('pDoc5',WtA(pDoc5,'C'));
arfn_confirm_doc._s.SetParameterString('pDoc6',WtA(pDoc6,'C'));
arfn_confirm_doc._s.SetParameterString('pOld0',WtA(pOld0,'C'));
arfn_confirm_doc._s.SetParameterString('pOld1',WtA(pOld1,'C'));
arfn_confirm_doc._s.SetParameterString('pOld2',WtA(pOld2,'C'));
arfn_confirm_doc._s.SetParameterString('pOld3',WtA(pOld3,'C'));
arfn_confirm_doc._s.SetParameterString('pOld4',WtA(pOld4,'C'));
arfn_confirm_doc._s.SetParameterString('pOld5',WtA(pOld5,'C'));
arfn_confirm_doc._s.SetParameterString('pOld6',WtA(pOld6,'C'));
arfn_confirm_doc._s.SetParameterString('pAttivi',WtA(pAttivi,'C'));
arfn_confirm_doc._s.SetParameterString('pTicket',WtA(pTicket,'C'));
arfn_confirm_doc._s.SetParameterString('pCodDip',WtA(pCodDip,'C'));
arfn_confirm_doc._s.SetParameterString('m_bApplet','true');
arfn_confirm_doc._s.CallServlet('arfn_confirm_doc');
if (Eq(typeof(m_bCalculating),'undefined') ||  ! (m_bCalculating)) {
m_cErrorFromRoutine=arfn_confirm_doc._s.errmsg;
}
return arfn_confirm_doc._s.GetString();
}
arfn_confirm_doc.CtxLoad_ = function(){return []}

File: JavaServletSrc/arfn_confirm_doc.java
import java.io.*;
import java.text.*;
import java.util.*;
import javax.servlet.*;
import javax.servlet.http.*;
import com.zucchetti.sitepainter.SPPrxycizer;

@com.zucchetti.sitepainter.EntityRoles({})
public class arfn_confirm_doc extends SPRoutineServlet implements SPInvokable {
  public static String m_cEntityUid = "!*f92a9def*!";
  public class ServletStatus extends SPRoutineServletStatus {
    public String gPathApp;
    public String cActLog;
    public double gUserCode;
    public String gAzienda;
    public String gPathDoc;
    public String _porig;
    public String _pdest;
    public String _olddoc;
    public String _prgdoc;
    public double _found;
    public boolean _creata;
    public String _nomedoc;
    public String _nomeorg;
    public String host;
    public String user;
    public String password;
    public String cRitorno;
    public String _completo;
    public String cProg;
    public String _errore;
    public String cOldUnico;
    public String cOldDoc1;
    public String cOldDoc2;
    public String cOldDoc3;
    public String cOldDoc4;
    public String cOldDoc5;
    public String cOldDoc6;
    public double _contadoc;
    public double _richiesti;
    public boolean _exist;
    public String xOld0;
    public String xOld1;
    public String xOld2;
    public String xOld3;
    public String xOld4;
    public String xOld5;
    public String xOld6;
    public boolean _validfld;
    public String _onamedoc;
    public String _extfile;
    public String _okm;
    public String pID;
    public String pTipo;
    public String pDocUnico;
    public String pDoc1;
    public String pDoc2;
    public String pDoc3;
    public String pDoc4;
    public String pDoc5;
    public String pDoc6;
    public String pOld0;
    public String pOld1;
    public String pOld2;
    public String pOld3;
    public String pOld4;
    public String pOld5;
    public String pOld6;
    public String pAttivi;
    public String pTicket;
    public String pCodDip;
    public String aaaaaaaaaaaaaaaaaaa_retval;
    public void GetNonLocals(SPParameterSource source) {
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pID","pID");
      }        
      pID = source.GetParameter("pID","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pTipo","pTipo");
      }        
      pTipo = source.GetParameter("pTipo","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pDocUnico","pDocUnico");
      }        
      pDocUnico = source.GetParameter("pDocUnico","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pDoc1","pDoc1");
      }        
      pDoc1 = source.GetParameter("pDoc1","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pDoc2","pDoc2");
      }        
      pDoc2 = source.GetParameter("pDoc2","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pDoc3","pDoc3");
      }        
      pDoc3 = source.GetParameter("pDoc3","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pDoc4","pDoc4");
      }        
      pDoc4 = source.GetParameter("pDoc4","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pDoc5","pDoc5");
      }        
      pDoc5 = source.GetParameter("pDoc5","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pDoc6","pDoc6");
      }        
      pDoc6 = source.GetParameter("pDoc6","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pOld0","pOld0");
      }        
      pOld0 = source.GetParameter("pOld0","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pOld1","pOld1");
      }        
      pOld1 = source.GetParameter("pOld1","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pOld2","pOld2");
      }        
      pOld2 = source.GetParameter("pOld2","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pOld3","pOld3");
      }        
      pOld3 = source.GetParameter("pOld3","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pOld4","pOld4");
      }        
      pOld4 = source.GetParameter("pOld4","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pOld5","pOld5");
      }        
      pOld5 = source.GetParameter("pOld5","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pOld6","pOld6");
      }        
      pOld6 = source.GetParameter("pOld6","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pAttivi","pAttivi");
      }        
      pAttivi = source.GetParameter("pAttivi","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pTicket","pTicket");
      }        
      pTicket = source.GetParameter("pTicket","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.pCodDip","pCodDip");
      }        
      pCodDip = source.GetParameter("pCodDip","");
    }
  }
  public SPEntityGlobals getEntityGlobals() {
    // Costanti globali dell' entita'
    SPRoutineGlobals l_eg;
    l_eg = new SPRoutineGlobals();
    l_eg.name = "arfn_confirm_doc";
    l_eg.hasProgressItem = false;
    l_eg.isPublic = false;
    return l_eg;
  }
  public void doProcess(HttpServletRequest request,HttpServletResponse response) throws IOException, ServletException {
    // Inizializzazione delle variabili della classe globale di stato
    ServletStatus status;
    status = new ServletStatus();
    status.containing = this;
    SPParameterSource source;
    source = SPLib.GetSource(request);
    if ( ! (EnterServlet(status,source,request,response))) {
      return;
    }
    Begin(status);
  }
  boolean NeedsFetchPage(ServletStatus status) {
    return false;
  }
  private void ExecuteRoutine(ServletStatus status) throws IOException {
    arfn_confirm_docR Rcalled;
    Rcalled = new arfn_confirm_docR(status.context,status);
    if (CPLib.ne(status.m_cAction,"executeAsync")) {
      ExecuteWorkflowStartingScript(status,Rcalled);
      status.aaaaaaaaaaaaaaaaaaa_retval = Rcalled.Run(status.pID,status.pTipo,status.pDocUnico,status.pDoc1,status.pDoc2,status.pDoc3,status.pDoc4,status.pDoc5,status.pDoc6,status.pOld0,status.pOld1,status.pOld2,status.pOld3,status.pOld4,status.pOld5,status.pOld6,status.pAttivi,status.pTicket,status.pCodDip);
      if (Rcalled.m_bError) {
        status.m_bError = true;
        status.m_cErrMsg = Rcalled.m_cLastMsgError;
      } else if ( ! (CPLib.Empty(status.m_cFaultTrace))) {
        status.m_bError = true;
        status.m_cErrMsg = status.m_cFaultTrace;
      }
      if (CPLib.eqr(CPLib.Left(status.m_cAtExit,8),"workflow")) {
        ExitToWorkflow(status);
      }
      if (Forward.IsForwarded(status.forwardedto)) {
        if (CPLib.IsNull(status.forwardedto.GetParameterString("m_cAtExit",null))) {
          status.forwardedto.SetParameter("m_cAtExit",status.m_cAtExit);
        }
        SPLib.ForwardTo(status.forwardedto,servletContext,status.request,status.response);
        return;
      }
    }
    BringBackPage(status,Rcalled);
  }
  CallerWithObjs NewRoutine(SPServletStatus status,Caller dataFromClient) {
    return new arfn_confirm_docR(status.context,dataFromClient);
  }
  protected void BringBackPage(ServletStatus status,arfn_confirm_docR Rcalled) throws IOException {
    String l_cErrMsg;
    String msg;
    msg = status.context.Translate("MSG_END_ROUTINE");
    SPLib.SetContentType(status.response);
    SPLib.NoCache(status.response);
    status.out = status.response.getWriter();
    String l_cJsUid;
    status.out.println(
     "<!doctype html>" +
    "");
    status.out.println(
     "<html>" +
    "");
    status.out.println(
     "<head>" +
    "");
    status.out.println(
     "<title>" +
     ""+status.context.Translate("MSG_ROUTINE_WIND")+"" +
     "</title>" +
    "");
    l_cJsUid = SPPrxycizer.proxycizedPath("stdFunctions.js");
    status.out.println(
     "<script src='../"+l_cJsUid+"' type='text/javascript'>" +
     "</script>" +
    "");
    l_cJsUid = SPPrxycizer.proxycizedPath("controls.js");
    status.out.println(
     "<script src='../"+l_cJsUid+"' type='text/javascript'>" +
     "</script>" +
    "");
    status.out.println(
     "<link type='text/css' rel='STYLESHEET' href='../"+status.m_cTheme+"/formPage.css'>" +
    "");
    status.out.println(
     "</head>" +
    "");
    if (status.m_bError) {
      msg = status.context.Translate("MSG_END_ROUTINE_WITH_MSG");
      status.out.println(
       "<body onkeydown='window.ManageKeys(event,false,false)'>" +
      "");
    } else {
      status.out.println(
       "<body>" +
      "");
    }
    status.out.println(
     "<p class='TabLabel' style='margin:0'>" +
     ""+msg+"" +
     "</p>" +
    "");
    if (status.m_bError) {
      status.out.println(
       "<script type='text/javascript'>" +
      "");
      l_cErrMsg = status.context.Translate(com.zucchetti.sitepainter.Library.explicativeErrorMessage(status.m_cErrMsg));
      status.out.println("detailed_database_error="+SPLib.ToJSValue(l_cErrMsg,"C",0,0)+";");
      status.out.println("m_cBrowser=" + SPLib.ToJSValue(status.m_cBrowser,"C",0,0)+ ";");
      status.out.println(
       "</script>" +
      "");
    }
    status.out.println(
     "<!--" +
    "");
    if ( ! (CPLib.Empty(Rcalled.m_cLastMsgError))) {
      CPTokenSource ts;
      ts = new CPTokenSource(status.context.Translate(com.zucchetti.sitepainter.Library.explicativeErrorMessage(Rcalled.m_cLastMsgError)),"\n");
      while (ts.HasMoreTokens()) {
        status.out.println("Error message:"+com.zucchetti.sitepainter.Library.ToHTMLEuro(ts.NextToken())+"");
      }
    }
    if ( ! (CPLib.Empty(status.m_cFaultTrace))) {
      status.out.println("Fault:["+SPLib.ToJSValue(status.context.Translate("MSG_ROUTINE_FAULTED","arfn_confirm_doc"),"C",0,0)+","+SPLib.ToJSValue(status.m_cFaultTrace,"C",0,0)+"]");
      if (CPLib.Empty(Rcalled.m_cLastMsgError) &&  ! (status.m_bApplet)) {
        status.out.println("Error message:"+com.zucchetti.sitepainter.Library.ToHTMLEuro(status.m_cFaultTrace)+"");
      }
    }
    status.PrintMutableParameters();
    status.out.println("Function return value:"+com.zucchetti.sitepainter.Library.ToHTMLEuro(CPLib.ToProperties(status.aaaaaaaaaaaaaaaaaaa_retval))+"");
    status.out.println(
     " -->" +
    "");
    status.out.println(
     "</body>" +
    "");
    status.out.println(
     "</html>" +
    "");
  }
  private void Begin(ServletStatus status) throws IOException {
    if ( ! (CPLib.Empty(status.m_cGlobalsToRead))) {
      JSGlobals(status);
    } else if (CPLib.Empty(status.m_cBrowseName)) {
      ExecuteRoutine(status);
    } else {
      JSResultSet(status);
    }
  }
  void IncludeJS(ServletStatus status) {
    /* --- Area Manuale = UI - Include */
    /* --- Fine Area Manuale */
  }
  // m_CallableNames inizializzato dal primo accesso a GetCallableNames
  private static volatile String[] m_CallableNames;
  public static String[] GetCallableNames() {
    if (CPLib.IsNull(m_CallableNames)) {
      java.util.Set building = new java.util.HashSet();
      GetCallableNames(building);
      m_CallableNames = CPLib.SetOfStringsAsArray(building);
    }
    return (String[])m_CallableNames.clone();
  }
  public static void GetCallableNames(java.util.Set callable) {
  }
  String[] RoutineIDChain() {
    return GetCallableNames();
  }
}
